src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -38,31 +54,40 @@ public sealed class PeerDiscoveryLoop : IDisposable
         /// <summary>Factory for creating P2P network peers.</summary>
         private readonly INetworkPeerFactory networkPeerFactory;
 
-        public PeerDiscoveryLoop(
-            IAsyncLoopFactory asyncLoopFactory,
-            Network network,
-            NetworkPeerConnectionParameters connectionParameters,
-            INodeLifetime nodeLifetime,
-            IPeerAddressManager peerAddressManager,
-            INetworkPeerFactory networkPeerFactory)
+        public PeerDiscoveryLoop(IConnectionManager connectionManager)
```

Comment:
Any reason why not to use injection? then you can prevent adding all the properties on the IConnectionManager interface.

Reply:
We now use DI to construct PeerDiscovery and inject it into the Connection Manager